### DISCUSSION TITLE
Add a remix project

### DISCUSSION THREAD

**Discussion by kjellaso** [TypeFox/monaco-languageclient#587] [General]
Hi! Anyone tried using the editor in a remix project? Is the best solution to setup this project as a standalone application, and then bring it in to my remix app with an iframe, or is it possible to integrate it with my remix app?

**Comment by msujew** [comment]
@kjellaso As far as I understand remix, it seems to be rendering just normal HTML. Did I get that right?

You should be able to just use a similar mechanism as the [browser example](https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/browser.html). Just create a simple `div` with an `id`:

https://github.com/TypeFox/monaco-languageclient/blob/789f8e5b5202817882bfb333d762c200edcb5235/packages/examples/browser.html#L13

And use that element as a container for the monaco editor:

https://github.com/TypeFox/monaco-languageclient/blob/789f8e5b5202817882bfb333d762c200edcb5235/packages/examples/src/browser/main.ts#L29

**Reply by kjellaso** [reply]
Thanks for the quick response! I think i found the solution. It seems can force Remix to only bundle to code to be run on client side, and then I can use the React component for Monaco-Editor and Monaco Languageclient. 

But it's a bit unclear from the documentation (or I'm just to rookie to figure it out) if I can use it as an npm package, or if I have to clone the repository and build it myself. Do you have any pointers for me @msujew ? 😊

**Reply by msujew** [reply]
It should be enough to just consume this package from npm. Have you encountered any issues with that approach?

**Reply by kjellaso** [reply]
No, seems I'm just jumping to conclusions. Just confused by the documentation that tells me to do "npm run build" before the example is presented . But I'll try to install the packages referenced in the example before wasting your time again 😊

**Reply by kaisalmen** [reply]
@kjellaso the examples in this project can rely on the local build if you check out the project (this is an npm workspace). Therefore it makes sense to build or watch the code before you run the examples if you changed code locally. The tagged commit are available on npm, e.g.: https://github.com/TypeFox/monaco-languageclient/tree/mlc-7.2.0

If you work in an independent project, just use the npm package.

**Reply by kaisalmen** [reply]
We have not deployed the examples somewhere (some rely on external server processes = node based language servers), but we could think about deploying the ones suitable to GitHub pages in the future. Then you could look at them without cloning the repo.

**Reply by kjellaso** [reply]
Thank you!
